gsa/src/stories/savedialog.js

```diff
@@ -0,0 +1,140 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import React from 'react';
+import {storiesOf} from '@storybook/react';
+import SaveDialog from '../web/components/dialog/savedialog';
+import Button from '../web/components/form/button';
+import TextField from '../web/components/form/textfield';
+
+class SaveButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: 'Save',
+      dialog: false,
+      error: '',
+      result: '',
+      input: '',
+      dialogValue: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleError = this.handleError.bind(this);
+    this.handleErrorClose = this.handleErrorClose.bind(this);
+  }
+
+  handleClick(value, name) {
+    this.setState({
+      dialog: true,
+    });
+
+    /* setTimeout(() => {
+      this.setState({input: 'bar'});
+    }, 2000); */
+  }
+
+  handleChange(value, name) {
+    this.setState({
+      input: value,
+    });
+  }
+
+  handleSave(data) {
+    if (data.input === '') {
+      this.setState({
+        error: 'Error! No Input!',
+      });
+    } else {
+      this.setState({
+        dialog: false,
+        result: data.input,
+        input: '',
+      });
+    }
+  }
+
+  handleClose(value, name) {
+    this.setState({
+      dialog: false,
+      error: '',
+    });
+  }
+
+  handleError(value, name) {
+    this.setState({
+      dialog: false,
+    });
+  }
+
+  handleErrorClose(value, name) {
+    this.setState({
+      error: '',
+    });
+  }
+
+  render() {
+    let dialog;
+    if (this.state.dialog === true) {
+      dialog = (
```

Comment:
Should be put into the return statement via https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator too